packages/terra-application-links/src/tabs/_Tab.jsx

```diff
@@ -22,43 +35,108 @@ const propTypes = {
   onTabClick: PropTypes.func,
 };
 
-const ApplicationTab = ({
-  onTabClick,
-  path,
-  text,
-  ...customProps
-}) => (
-  <Route
-    render={({ location, history }) => {
-      const isActive = !!matchPath(location.pathname, { path });
-      const tabClassNames = cx([
-        'tab',
-        customProps.className,
-      ]);
-      const tabAttr = { 'aria-current': `${isActive}`, role: 'tab' };
-
-      return (
-        <button
-          {...customProps}
-          {...tabAttr}
-          className={tabClassNames}
-          onClick={(event) => {
-            if (!isActive) {
-              history.push(path);
-            }
-            if (onTabClick) {
-              onTabClick(event);
-            }
-          }}
-        >
-          <span className={cx(['tab-inner'])}>
-            {text}
-          </span>
-        </button>
-      );
-    }}
-  />
-);
+class ApplicationTab extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { active: false, focused: false };
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+    this.isCurrentPath = this.isCurrentPath.bind(this);
+  }
+
+  handleOnBlur() {
+    this.setState({ focused: false });
+  }
+
+  handleKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE) {
+      this.setState({ active: true });
+    }
+
+    // Add focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE || event.nativeEvent.keyCode === TabUtils.KEYCODES.ENTER) {
+      this.setState({ focused: true });
+
+      event.preventDefault();
+      this.handleOnClick(event);
+    }
+  }
+
+  handleKeyUp(event) {
+    // Remove active state from FF broswers
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.SPACE) {
+      this.setState({ active: false });
+    }
+
+    // Apply focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === TabUtils.KEYCODES.TAB) {
+      event.preventDefault();
```

Comment:
bump

Reply:
Without this logic the event is passed to the "more" button and the popup is re-displayed.